src/servicesHost.ts

```diff
@@ -27,8 +27,17 @@ function makeServicesHost(
     const resolveSync = makeResolver(loader.options);
 
     const moduleResolutionHost = {
-        fileExists: (fileName: string) => utils.readFile(fileName) !== undefined,
-        readFile: (fileName: string) => utils.readFile(fileName),
+        fileExists: (fileName: string) => {
```

Comment:
When a file is generated by a previous loader in the chain, it does not necessarily exists in the file system. So to make sure the TypeScript compiler can resolve the file, I look up in `intance.files` first. Otherwise it would not be able to resolve generated files. This should actually give a slight performance improvement, as files will only be loaded once (and of course maybe in `watch-run`).